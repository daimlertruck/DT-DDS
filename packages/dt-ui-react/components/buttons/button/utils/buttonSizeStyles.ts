import { CustomTheme as Theme } from '../../../../types';
import { ButtonSizes } from '../constants/types';

export const buttonSizeStyles = (size: ButtonSizes, theme: Theme) =>
  ({
    large: `
      ${theme.fontStyles.button1}
      padding: ${theme.spacing['2xs']} ${theme.spacing.s};
    `,
    medium: `
      ${theme.fontStyles.button2}
      padding: ${theme.spacing['3xs']} ${theme.spacing.xs};
    `,
    small: `
      ${theme.fontStyles.button2}
      padding: ${theme.spacing['4xs']} ${theme.spacing['2xs']};
    `,
  }[size]);
