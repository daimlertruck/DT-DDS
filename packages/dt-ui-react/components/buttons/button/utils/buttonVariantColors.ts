import { CustomTheme as Theme } from '../../../../types';
import {
  ButtonVariant,
  ContainedColors,
  OutlinedColors,
  TextColors,
} from '../constants/types';

interface ColorKeys {
  color: string;
  bgColor: string;
  hoverBgColor?: string;
  hoverColor?: string;
}

export const buttonVariantColors = (
  color: ContainedColors | OutlinedColors,
  theme: Theme
): Record<ButtonVariant, ColorKeys> => ({
  contained: {
    color: theme.colors.alpha.white,
    bgColor: theme.palette.alpha[color],
    hoverBgColor: theme.palette.alpha[`${color}Dark_300` as ContainedColors],
  },
  outlined: {
    color:
      color === 'neutral'
        ? theme.colors.alpha.gray_200
        : theme.palette.alpha[color],
    bgColor: theme.colors.alpha.white,
    hoverBgColor: theme.palette.alpha[`${color}Light_300` as OutlinedColors],
  },
  text: {
    color: theme.palette.alpha[color],
    bgColor: 'transparent',
    hoverColor: theme.palette.alpha[`${color as TextColors}Dark_300`],
  },
});
