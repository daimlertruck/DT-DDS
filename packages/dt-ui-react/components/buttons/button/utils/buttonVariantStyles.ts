import { CustomTheme as Theme } from '../../../../types';
import {
  ButtonVariant,
  ContainedColors,
  OutlinedColors,
} from '../constants/types';

export const buttonVariantStyles = (
  variant: ButtonVariant,
  theme: Theme,
  color: ContainedColors | OutlinedColors
) =>
  ({
    outlined: `
      border: 1px solid;
      text-transform: uppercase;
        ${
          color === 'neutral'
            ? theme.palette.alpha.neutralDark_100
            : theme.palette.alpha[color]
        };

      &:disabled {
        color: ${
          color === 'neutral'
            ? theme.palette.alpha.neutralLight_100
            : theme.palette.alpha[color]
        };
        background-color: ${theme.colors.white};
        border-color: ${
          color === 'neutral'
            ? theme.palette.alpha.neutralLight_100
            : theme.palette.alpha[color]
        };
        cursor: default;
      }
    `,
    contained: `
      border: ${theme.spacing.none};
      text-transform: uppercase;
      &:disabled {
        opacity: 0.4;
        cursor: default;
      }
    `,
    text: `
      border: none;

      &:disabled {
        color: ${theme.palette.alpha.neutralLight_100};
      }
    `,
  }[variant]);
