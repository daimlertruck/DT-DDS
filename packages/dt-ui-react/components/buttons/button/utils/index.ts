export * from './buttonVariantColors';
export * from './buttonVariantStyles';
