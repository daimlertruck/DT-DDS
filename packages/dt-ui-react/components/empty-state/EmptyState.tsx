import { BaseProps } from '../../types';
import { Typography } from '../typography';

import {
  EmptyStateStyled,
  EmptyStateDescriptionStyled,
  EmptyStateTitleStyled,
} from './EmptyState.styled';

export type EmptyStateProps = {
  title: string;
  description: string;
} & BaseProps;

export const EmptyState = ({
  children,
  dataTestId,
  title,
  description,
}: EmptyStateProps) => (
  <EmptyStateStyled data-testid={dataTestId ?? 'empty-state'}>
    <EmptyStateTitleStyled>
      <Typography element='h2' fontStyles='h4' color='textPrimary'>
        {title}
      </Typography>
    </EmptyStateTitleStyled>
    <EmptyStateDescriptionStyled>
      <Typography fontStyles='body1' color='textSecondary'>
        {description}
      </Typography>
    </EmptyStateDescriptionStyled>
    {children}
  </EmptyStateStyled>
);
