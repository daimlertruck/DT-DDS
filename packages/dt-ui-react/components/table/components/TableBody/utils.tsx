import { Children, ReactNode, cloneElement, isValidElement } from 'react';

import { TableDataCellHeader } from '../TableDataCell/TableDataCell';

export const addHeadersToDataCells = (
  dataCells: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(dataCells, (dataCell, index) => {
    if (isValidElement(dataCell)) {
      const header = headers[index];
      return cloneElement(
        dataCell,
        {},
        <>
          <TableDataCellHeader>{header}</TableDataCellHeader>
          {dataCell.props.children}
        </>
      );
    }
    return dataCell;
  });
};

export const addHeadersToDataCellsInRows = (
  rows: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(rows, (row) => {
    if (isValidElement(row)) {
      return cloneElement(
        row,
        {},
        addHeadersToDataCells(row.props.children, headers)
      );
    }
    return row;
  });
};
