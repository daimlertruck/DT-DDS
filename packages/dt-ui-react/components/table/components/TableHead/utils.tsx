import { Children, ReactNode, isValidElement } from 'react';

const extractColumnHeaders = (columnHeaders: ReactNode): ReactNode[] => {
  return Children.toArray(columnHeaders).reduce<ReactNode[]>(
    (headers, columnHeader) => {
      if (isValidElement(columnHeader)) {
        return [...headers, columnHeader.props.children];
      }

      return headers;
    },
    []
  );
};

export const extractTableHeaders = (theadRow: ReactNode) => {
  if (isValidElement(theadRow)) {
    return extractColumnHeaders(theadRow.props.children);
  }

  return [];
};
