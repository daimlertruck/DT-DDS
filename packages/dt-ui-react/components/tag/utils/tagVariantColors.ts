import { CustomTheme as Theme } from '../../../types';
import { TagColor, TagVariant } from '../constants';

interface TagVariantColors {
  border: string | number;
  color: string;
  borderColor?: string;
  backgroundColor: string;
  '&:hover'?: {
    borderColor?: string;
    backgroundColor: string;
    color: string;
  };
}

interface TagVariantColorsProps {
  theme: Theme;
  color: TagColor;
  variant: TagVariant;
  hasHover: boolean;
  isDisabled: boolean;
}

const getColorWithShade = ({
  theme,
  color,
  shade,
}: Pick<TagVariantColorsProps, 'color'> & {
  shade: '90' | '60' | '30' | '10';
  theme: Theme;
}) => {
  const { colors, palette } = theme;

  if (color === 'primary') {
    return {
      '90': palette.primaryDark_500,
      '60': palette.primaryDark_200,
      '30': palette.primaryLight_100,
      '10': palette.primaryLight_400,
    }[shade];
  }

  const colorWithShade: keyof Theme['colors'] = `${color}_${shade}`;

  return colors[colorWithShade];
};

export const tagVariantColors = ({
  theme,
  color,
  variant,
  hasHover,
  isDisabled,
}: TagVariantColorsProps): TagVariantColors => {
  const colorShade90 = getColorWithShade({
    theme,
    color,
    shade: '90',
  });
  const colorShade60 = getColorWithShade({
    theme,
    color,
    shade: '60',
  });
  const colorShade30 = getColorWithShade({
    theme,
    color,
    shade: '30',
  });
  const colorShade10 = getColorWithShade({
    theme,
    color,
    shade: '10',
  });

  const colorWithDisabled = isDisabled ? colorShade30 : colorShade60;

  return {
    solid: {
      border: theme.spacing.none,
      color: theme.colors.white,
      backgroundColor: colorWithDisabled,
      ...(hasHover && {
        '&:hover': {
          backgroundColor: colorShade90,
          color: theme.colors.white,
        },
      }),
    },
    outlined: {
      border: `1px solid ${colorShade60}`,
      color: colorWithDisabled,
      backgroundColor: colorShade10,
      borderColor: colorWithDisabled,
      ...(hasHover && {
        '&:hover': {
          borderColor: colorShade90,
          backgroundColor: colorShade10,
          color: colorShade90,
        },
      }),
    },
  }[variant];
};
