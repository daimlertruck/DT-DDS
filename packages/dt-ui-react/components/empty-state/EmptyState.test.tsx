import { render } from '@testing-library/react';
import React from 'react';

import { withProviders } from '../../utils';
import { Button } from '../buttons';

import { EmptyState } from './EmptyState';

describe('<EmptyState /> component', () => {
  const ProvidedEmptyState = withProviders(EmptyState);
  const title = 'Title here';
  const description = 'Some description here.';

  it('should render Empty State', () => {
    const { container } = render(
      <ProvidedEmptyState title={title} description={description} />
    );

    expect(container).toMatchSnapshot();
  });

  it('should render Empty State with a Button as a children', () => {
    const { container } = render(
      <ProvidedEmptyState title={title} description={description}>
        <Button onClick={() => {}} variant='contained' dataTestId='empty-state'>
          Button
        </Button>
      </ProvidedEmptyState>
    );

    expect(container).toMatchSnapshot();
  });
});
