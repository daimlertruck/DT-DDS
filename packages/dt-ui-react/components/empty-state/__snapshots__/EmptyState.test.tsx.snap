// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyState /> component should render Empty State 1`] = `
.emotion-0 {
  padding: 48px 32px;
  background-color: #F8F9FA;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 8px;
  border: 1px dashed #D3D4D5;
  width: 100%;
  height: 100%;
}

@media only screen and (min-width: 600px) {
  .emotion-0 {
    padding: 64px 128px;
  }
}

.emotion-2 {
  text-align: center;
}

.emotion-4 {
  font-weight: 400;
  font-size: 2rem;
  line-height: 1.23;
  margin: 0;
  color: #323232;
}

.emotion-6 {
  text-align: center;
  padding-top: 8px;
  padding-bottom: 0;
}

@media only screen and (min-width: 600px) {
  .emotion-6 {
    padding-top: 12px;
  }
}

.emotion-8 {
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #7C7C7D;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-testid="empty-state"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <h2
        class="emotion-4 emotion-5"
        color="textPrimary"
        data-testid="typography"
      >
        Title here
      </h2>
    </div>
    <div
      class="emotion-6 emotion-7"
    >
      <p
        class="emotion-8 emotion-5"
        color="textSecondary"
        data-testid="typography"
      >
        Some description here.
      </p>
    </div>
  </div>
</div>
`;

exports[`<EmptyState /> component should render Empty State with a Button as a children 1`] = `
.emotion-0 {
  padding: 48px 32px;
  background-color: #F8F9FA;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 8px;
  border: 1px dashed #D3D4D5;
  width: 100%;
  height: 100%;
}

@media only screen and (min-width: 600px) {
  .emotion-0 {
    padding: 64px 128px;
  }
}

.emotion-2 {
  text-align: center;
}

.emotion-4 {
  font-weight: 400;
  font-size: 2rem;
  line-height: 1.23;
  margin: 0;
  color: #323232;
}

.emotion-6 {
  text-align: center;
  padding-top: 8px;
  padding-bottom: 24px;
}

@media only screen and (min-width: 600px) {
  .emotion-6 {
    padding-top: 12px;
  }
}

.emotion-8 {
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #7C7C7D;
}

.emotion-10 {
  border: 0;
  text-transform: uppercase;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.34;
  margin: 0;
  font-weight: 700;
  border-radius: 3px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
  white-space: nowrap;
  cursor: pointer;
  max-width: 20vw;
  padding: 0px 15px;
  height: 32px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  color: #FFFFFF;
  background-color: #0078D6;
}

.emotion-10:disabled {
  opacity: 0.4;
  cursor: default;
}

.emotion-10:disabled {
  pointer-events: none;
}

.emotion-10:hover {
  color: #FFFFFF;
  background-color: #01508E;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-testid="empty-state"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <h2
        class="emotion-4 emotion-5"
        color="textPrimary"
        data-testid="typography"
      >
        Title here
      </h2>
    </div>
    <div
      class="emotion-6 emotion-7"
    >
      <p
        class="emotion-8 emotion-5"
        color="textSecondary"
        data-testid="typography"
      >
        Some description here.
      </p>
    </div>
    <button
      class="emotion-10 emotion-11"
      color="primary"
      data-testid="empty-state-button"
    >
      Button
    </button>
  </div>
</div>
`;
